### ISSUE TITLE
Edge cases in generated instructions: dual-scope option drift and limited virtual file sampling

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#122]
## Summary

During review of #121, two edge cases were identified in the generated instructions feature that were deferred from the initial implementation. Tracking here for future prioritization.

## Edge Case 1: TS-specific options lost for dual-scope rules

**File:** `src/instructions/config-resolver.ts`, line 149

When a rule is enabled in both JS and TS configs with **different option values**, the resolver always picks JS options for interpolation while marking the instruction as applying to `all` files. This produces incorrect guidance for TypeScript files.

**Example:**
```js
// eslint.config.mjs
rules: {
  "llm-core/max-function-length": ["error", { max: 50 }],  // JS
}
// overrides for .ts files:
rules: {
  "llm-core/max-function-length": ["error", { max: 30 }],  // TS
}
```

Generated instruction would say "Keep functions under 50 lines" for **all** files, but TypeScript files actually enforce 30.

**Possible fix:** When a rule is enabled in both scopes with differing options, emit two separate instructions — one per scope — each with the correct interpolated values.

## Edge Case 2: Only `.js` and `.ts` virtual files sampled

**File:** `src/instructions/config-resolver.ts`, lines 114-118

Scope derivation samples ESLint config with only `__virtual__.js` and `__virtual__.ts`. Rules configured exclusively for other supported script extensions (`.tsx`, `.jsx`, `.mjs`, `.cjs`) are never observed. In those configurations, active `llm-core` rules can be omitted or mis-scoped in generated output.

**Possible fix:** Expand virtual file sampling to include `.tsx`, `.jsx`, `.mjs`, `.cjs` (or a representative subset), merging results into broader "JS" and "TS" buckets. Consider the performance impact of additional `calculateConfigForFile` calls.

## Context

- PR: #121
- Review source: Codex automated review
- Severity: P2 (edge cases, not blocking for initial release)

**Comment by snakefood3232** [comment]
The dual-scope option drift and limited virtual file sampling for .tsx/.jsx are clear issues. I'd handle this by modifying the config resolver to emit separate instructions per scope when options differ and expanding the virtual file sampling to all supported extensions. I can have a PR up in a day—I've built similar ESLint config analysis tooling before.

**Comment by snakefood3232** [comment]
The dual-scope option drift and limited virtual file sampling are clear edge cases. I'd handle this by generating separate instructions per scope when options differ and expanding the virtual file sampling to include all supported extensions. Can have a PR up in a day—I've built similar ESLint config analyzers before.
